File: file_processing/pl/lte_threshold.pl
#!/usr/bin/perl
## prints numbers in a file in $ARGV[0] that are <= $ARV[1]
## to run: >perl lte_thresh.pl numbers.txt 10
## @author: vladimir kulyukin
use strict;
use warnings;
my $file_path = shift();
my $thresh    = shift();
open(my $infile, '<', $file_path) or die $!;
while ( <$infile> ) { print $_ if ( $_ <= $thresh ); }

File: file_processing/py/lte_threshold.py
#!/usr/bin/python
## print numbers from file in sys.argv[1] that are <= int(sys.argv[2])
## to run: >python lte_thresh.py numbers.txt 10
## @author: vladimir kulyukin
import sys
file_path, thresh = sys.argv[1], int(sys.argv[2])
with open(file_path, 'r') as infile:
    for n in [int(line) for line in infile.readlines()
              if int(line) <= thresh]:
        print n
